clcache.py

```diff
@@ -748,13 +748,17 @@ def _preprocessToStdout(options):
 
     @staticmethod
     def _parseOptionsAndFiles(cmdline):
-        optionsWithParameter = ['Ob', 'Gs', 'Fa', 'Fd', 'Fm',
-                                'Fp', 'FR', 'doc', 'FA', 'Fe',
-                                'Fo', 'Fr', 'AI', 'FI', 'FU',
-                                'D', 'U', 'I', 'Zp', 'vm',
-                                'MP', 'Tc', 'V', 'wd', 'wo',
-                                'W', 'Yc', 'Yl', 'Tp', 'we',
-                                'Yu', 'Zm', 'F', 'Fi']
+        # sorted by length to handle prefixes
+        optionsWithParameter = [
+            # length 3
+            'doc',
+            # length 2
+            'AI', 'Fa', 'FA', 'Fd', 'Fe', 'Fi', 'FI', 'Fm', 'Fo',
+            'Fp', 'Fr', 'FR', 'FU', 'Gs', 'MP', 'Ob', 'Tc', 'Tp',
+            'vm', 'wd', 'we', 'wo', 'Yc', 'Yl', 'Yu', 'Zm', 'Zp',
+            # length 1
+            'D', 'F', 'I', 'U', 'V', 'W',
+        ]
```

Comment:
Good catch! However, I think it would be preferable to rather just add the line

```python
optionsWithParameter.sort(key=len, reverse=True)
```

The diff is smaller, and it's harder to break this in case we need to extend the list of options.